### ISSUE TITLE
Inference speed slows down after using graph transform tool

### ISSUE THREAD

**Issue by yoyolin** [tensorflow/tensorflow#25495]
**System information**
- Have I written custom code: No, I used tensorflow's object detection api's officially released model `ssd-mobilenet-v1-fpn`.
- OS Platform and Distribution: Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): built from source, with xla and mkl support, use nvcc as compiler.
- TensorFlow version (use command below): r1.12.0
- Python version: Python 3.6.4 |Anaconda
- Bazel version (if compiling from source): 0.18.1
- GCC/Compiler version (if compiling from source): 5.4.0
- CUDA/cuDNN version: CUDA 9.0, cuDNN 7.4.2
- GPU model and memory: 3*NVIDIA 1080Ti, 

**Describe the current behavior**
I trained `ssd-mobilenet-v1-fpn` from tensorflow object detection api with default configurations. As its inference was pretty slow on CPU, I checked using tensorflow's profiler advise and found out the main time consumption was from batch normalization(took 140 ms in total).  Therefore, I tried to use [tensorflow's graph transform tool](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/graph_transforms) to make if faster.

But the inference speed actually slowed down after the graph transformation. To be specific, inference takes 
* 0.53 seconds for original inference graph
* 0.56 seconds for optimized inference graph

Why is the case?

An additional information is that the size actually increases too: the original inference graph is 43MB and optimized inference graph is 44MB. Not sure it will be useful.

Here is the code I used:
```
bazel-bin/tensorflow/tools/graph_transforms/transform_graph \
--in_graph=fpnssd/frozen_inference_graph.pb \
--out_graph=fpnssd/optimized_frozen_inference_graph.pb \
--inputs="image_tensor:0" \
--outputs="detection_boxes:0,detection_scores:0,detection_classes:0,num_detections:0" \
--transforms="strip_unused_nodes() fold_constants(ignore_errors=false) fold_batch_norms fold_old_batch_norms"
```

**Describe the expected behavior**
I expect it to be quicker with optimized graph.

**Code to reproduce the issue**
```
bazel-bin/tensorflow/tools/graph_transforms/transform_graph \
--in_graph=fpnssd/frozen_inference_graph.pb \
--out_graph=fpnssd/optimized_frozen_inference_graph.pb \
--inputs="image_tensor:0" \
--outputs="detection_boxes:0,detection_scores:0,detection_classes:0,num_detections:0" \
--transforms="strip_unused_nodes() fold_constants(ignore_errors=false) fold_batch_norms fold_old_batch_norms"
```

**Comment by yoyolin** [comment]
Also used graph-transform on `deeplab-xception_65` and below is a comparison using tensorflow's benchmarking tool.

|   | original graph  | optimized graph   |
|---|---|---|
|  Timings (microseconds) |avg=120329   |  avg=112269 |
|  Memory (bytes)   |  avg=3.79e+09  |  curr=3789380632(all same)  |
| nodes observed  |  1559  |  1337 |
|  FLOPs estimate | 337.22B  | 337.22B   |
|  FLOPs/second  | 3.35T   |  3.55T |

Seems to work this time.  

The graph transform command I used is 

```
bazel-bin/tensorflow/tools/graph_transforms/transform_graph \
  --in_graph=$GRAPH_PATH/frozen_inference_graph.pb \
  --out_graph=$GRAPH_PATH/optimized_inference_graph.pb \
  --inputs="ImageTensor:0" \
  --outputs="SemanticPredictions:0" \
  --transforms='remove_device strip_unused_nodes(type=uint8, shape="-1,401,531,3") remove_nodes(op=CheckNumerics, op=Identity) fold_constants(ignore_errors=true) fold_batch_norms fold_old_batch_norms'
```

**Comment by austingg** [comment]
@yoyolin  you may try to add 'fold_old_batch_norm' transform for xception_65, that will get kind of speed up, maybe 5~10%. However,  when it comes to ssd-mobilenet-fpn, the transform graph tool may broken the model, which lead to wrong result, and I'm working on it.

**Comment by rdzhabarov** [comment]
@austingg @yoyolin have you guys been able to get to the bottom of it? Does this issue still persist?

**Comment by tilakrayal** [comment]
@yoyolin,
Could you please try The TFG transform tool  which will  be good to use which supports **graphdef in - graphdef out**.

And also looks like that you are trying to apply the passes to the graphdef which might not have an equivalent in TFG. 
https://www.tensorflow.org/guide/graph_optimization
https://discuss.tensorflow.org/t/design-of-mlir-tfg-tensor/6371/3

Thank you!

**Comment by tilakrayal** [comment]
Closing this as stale. Please reopen if this is still a valid request. Thank you!
